src/PyDynamic/misc/tools.py

```diff
@@ -160,6 +160,92 @@ def trimOrPad(
         return array[0:length]
 
 
+def trimOrPad_ND(
+    array: Union[List, np.ndarray],
+    length: Union[int, tuple],
+    mode: Optional[str] = "constant",
+    real_imag_type=False,
+):
```

Comment:
As this is a more powerful method fully covering the use case of trimOrPad and allows the same inputs as trimOrPad does, even producing the same results then, we should replace trimOrPad with this implementation under the name trimOrPad. To introduce a Changelog entry with the new capabilities, a commit of type `feat` should be added, when doing this refactoring. At this time, the module docstring (line 23 at the moment) stating 'trim or pad (with zeros) a vector to desired length' should be adapted accordingly.

Reply:
Resolved by 1d4acfc.